src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -569,62 +631,95 @@ private async Task ProcessSubscriptionAsync()
-                                    subscriptionChangeVectorBeforeCurrentBatch);
+                                        subscriptionChangeVectorBeforeCurrentBatch);
 
-                                subscriptionChangeVectorBeforeCurrentBatch = _lastChangeVector ?? SubscriptionState.ChangeVectorForNextBatchStartingPoint;
+                                    subscriptionChangeVectorBeforeCurrentBatch = _lastChangeVector ?? SubscriptionState.ChangeVectorForNextBatchStartingPoint;
 
-                                if (sendingCurrentBatchStopwatch.ElapsedMilliseconds > 1000)
-                                    await SendHeartBeat("Didn't find any documents to send and more then 1000ms passed");
+                                    UpdateBatchPerformanceStats(false);
+                                    
+                                    if (sendingCurrentBatchStopwatch.ElapsedMilliseconds > 1000)
+                                        await SendHeartBeat("Didn't find any documents to send and more then 1000ms passed");
+                                    
+                                    using (docsContext.OpenReadTransaction())
+                                    {
+                                        var globalEtag = useRevisions
+                                            ? TcpConnection.DocumentDatabase.DocumentsStorage.RevisionsStorage.GetLastRevisionEtag(docsContext, Subscription.Collection)
+                                            : TcpConnection.DocumentDatabase.DocumentsStorage.GetLastDocumentEtag(docsContext.Transaction.InnerTransaction, Subscription.Collection);
 
-                                using (docsContext.OpenReadTransaction())
-                                {
-                                    var globalEtag = useRevisions ?
-                                        TcpConnection.DocumentDatabase.DocumentsStorage.RevisionsStorage.GetLastRevisionEtag(docsContext, Subscription.Collection) :
-                                        TcpConnection.DocumentDatabase.DocumentsStorage.GetLastDocumentEtag(docsContext.Transaction.InnerTransaction, Subscription.Collection);
+                                        if (globalEtag > _startEtag)
+                                            continue;
+                                    }
 
-                                    if (globalEtag > _startEtag)
+                                    AssertCloseWhenNoDocsLeft();
+                                    
+                                    if (await WaitForChangedDocuments(replyFromClientTask)) 
                                         continue;
                                 }
+                            }
 
-                                AssertCloseWhenNoDocsLeft();
-
-                                if (await WaitForChangedDocuments(replyFromClientTask))
-                                    continue;
+                            using (batchParentScope.For(SubscriptionOperationScope.Ack))
```

Comment:
we can probably rename batchParentScope => batchScope

Reply:
imho - suggesting to just rename to: `parentBatchScope`
because at the time we called the connection scope variable: `_parentConnectionScope`